content/docs/use-cases/ci-and-ml.md

```diff
@@ -0,0 +1,171 @@
+data processing steps of a pipeline.
+
+Here is a non exhaustive list of unique challenges:
+
+- Automatically re-train models upon changes.
+- Compare (and report) performance of different versions of a model.
+- Test data correctness when new data is introduced.
+- Model training is computationally intensive.
+
+Combining a version control tool (like `git`) together with `dvc` and `cml`
+opens a wide spectrum of possibilities that can help building better and smarter
+data products. Adding [CI](https://en.wikipedia.org/wiki/Continuous_integration)
+concepts to this triple allows the bulding and **coding** of flows that can help
+tackling the aforementioned problems and many others. In this article, you will
+learn about a few CI use cases utilizing `dvc` and `cml`.
+
+_Remarks:_ The code snippets in this article are for demonstration purposes and
+not actually running snippets. Furthermore, it is assumed that the `git`
+repository is hosted on `GitHUb`; the same ideas can be implemented when using
+`GitLab` or any other hosting solution that suports CI.
+
+## Automatic model training
+
+Training a model is often a very demanding process. Nevertheless, it is very
+much desired to re-train whenever new data is available and/or changes were made
+to the model itself. Similarly, a test suite of a project is very involved and
+complex. CI is used to perform the testing; let it be on various platforms and
+environments, or simply move the testing load from a developer’s local machine
+to a dedicated testing platform.
+
+![](/img/ci_for_ml_retrain_model.jpg) _Model training flow_
+
+Assuming that in a project’s repository a `dvc` pipeline is defined for training
+a model. You can ask your CI tool to call `dvc pull` and, voilà, it is now has
+access to the _right_ data needed for training. The reproducing pipeline can be
+arbitrary complicated and comprise of many steps; regardless of all, `dvc repro`
+will yield a newly trained model. This new model will be tracked and pushed to
+the remote cache (note `dvc push`) and will be "linked" to the PR.
+
+```yml
+# .github/workflows/train_model.yml
+
+name: Retrain model
+on:
+  pull_request:
+    types: synchronize
+
+jobs:
+  train:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up the environment on the runner
+        run: |
+          apt-get update -y
+          apt-get install python3-dev -y
+          pip install -r requirements.txt
+
+      - name: DVC fetch the data used by the branch
+        run: dvc pull
+
+      - name: DVC retrain the model
+        run: dvc repro
+
+      - name: Track the newly trained model
+        run: |
+          dvc add model.pkl
+          git config --global user.name 'Your Name'
+          git config --global user.email 'my@enail.com'
+          git add model.pkl.dvc
+          git commit -m "Model retrained upon sync to PR"
+          git push
+          dvc phsh
```

Comment:
I am not sure whether the details here work as I expect them to. But I do hope the idea is clear.

Reply:
Typo: `phsh` 🙂 

BTW what language is this? Github CI/CD? Maybe worth mentioning.

Reply:
Typo indeed! Will fix. Thanks.

That's a YMAL for GitHub Actions. I do mention at the beginning that the snippets are only for demonstration purposes and assume usage of GitHub. I based on https://cml.dev/.

Otherwise? @jorgeorpinel feel free to resolve this.

Reply:
I see the Remarks now, thanks.